sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClientBuilder.java

```diff
@@ -0,0 +1,221 @@
+     * version based on the version of the client library being used. If no service version is specified, updating to a
+     * newer version of the client library will have the result of potentially moving to a newer service version.
+     * <p>
+     * Targeting a specific service version may also mean that the service will return an error for newer APIs.
+     *
+     * @param serviceVersion The service API version to use.
+     * @return the updated DigitalTwinsClientBuilder instance for fluent building.
+     */
+    public DigitalTwinsClientBuilder serviceVersion(DigitalTwinsServiceVersion serviceVersion) {
+        this.serviceVersion = serviceVersion;
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpClient} to use for sending a receiving requests to and from the service.
+     *
+     * @param httpClient HttpClient to use for requests.
+     * @return the updated DigitalTwinsClientBuilder instance for fluent building.
+     */
+    public DigitalTwinsClientBuilder httpClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpLogOptions} for service requests.
+     *
+     * @param logOptions The logging configuration to use when sending and receiving HTTP requests/responses.
+     * @return the updated DigitalTwinsClientBuilder instance for fluent building.
+     * @throws NullPointerException If {@code logOptions} is {@code null}.
+     */
+    public DigitalTwinsClientBuilder httpLogOptions(HttpLogOptions logOptions) {
+        this.logOptions = Objects.requireNonNull(logOptions, "'logOptions' cannot be null.");
+        return this;
+    }
+
+    /**
+     * Adds a pipeline policy to apply on each request sent. The policy will be added after the retry policy. If
+     * the method is called multiple times, all policies will be added and their order preserved.
+     *
+     * @param pipelinePolicy a pipeline policy
+     * @return the updated DigitalTwinsClientBuilder instance for fluent building.
+     * @throws NullPointerException If {@code pipelinePolicy} is {@code null}.
+     */
+    public DigitalTwinsClientBuilder addPolicy(HttpPipelinePolicy pipelinePolicy) {
+        this.additionalPolicies.add(Objects.requireNonNull(pipelinePolicy, "'pipelinePolicy' cannot be null"));
+        return this;
+    }
+
+    /**
+     * Sets the request retry options for all the requests made through the client. By default, the pipeline will
+     * use an exponential backoff retry value as detailed in {@link RetryPolicy#RetryPolicy()}.
+     *
+     * @param retryPolicy {@link RetryPolicy}.
+     * @return the updated DigitalTwinsClientBuilder instance for fluent building.
+     * @throws NullPointerException If {@code retryOptions} is {@code null}.
+     */
+    public DigitalTwinsClientBuilder retryOptions(RetryPolicy retryPolicy) {
+        this.retryPolicy = Objects.requireNonNull(retryPolicy, "'retryPolicy' cannot be null.");
```

Comment:
The other code in the builder allows for `retryPolicy` to be `null`. I would remove the `null` check and just document that if this is `null` a default `RetryPolicy` will be used.

Reply:
I will make code changes after the first round of team review, I will make a note of this comment.